library/L2_HAL/communication/test/esp8266_test.cpp

```diff
@@ -60,15 +161,167 @@ TEST_CASE("Esp8266 Test", "[esp8266]")
+        CHECK(timeout == Esp8266::kReadTimeout);
+        CHECK(size == 128);
+
+        for (size_t i = 0; i < (sizeof(kTest)-1); i++)
+        {
+          data[i] = kTest[i];
+        }
+
+        count++;
+      }
+      // Checks FlushSerialBuffer has right timeout and size
+      else
+      {
+        CHECK(timeout == Esp8266::kReadTimeout);
+        CHECK(size == 1);
+      }
+
+      return Status::kTimedOut;
+    };
+    When(ConstOverloadedMethod(
+             mock_uart,
+             Read,
+             Status(uint8_t *, size_t, std::chrono::microseconds)))
+        .AlwaysDo(uart_response);
+    Esp8266::ModuleVersion_t version = test_wifi_module.GetModuleVersion();
+    // Checks to see if module_info in ModuleVersion_t struct was written to
+    for (size_t i = 0; i < sizeof(kTest); i++)
+    {
+      CHECK(version.module_info[i] == kTest[i]);
+    }
+    // Checks to see if GetModuleVersionCommand was written to the esp8266
+    Verify(
+        ConstOverloadedMethod(mock_uart, Write, void(const uint8_t *, size_t))
+            .Using(Esp8266::kGetModuleVersionCommand,
+              (sizeof(Esp8266::kGetModuleVersionCommand)-1)));
   }
+
   SECTION("Get Network Connection")
   {
+    static constexpr uint8_t kTest[] = "test\r\nOK";
+    // Used to read kTest to buffer when using uart read
+    auto uart_response =
+        [](uint8_t * data,
+           size_t size,
+           std::chrono::microseconds timeout) -> Status {
+      static uint8_t count = 0;
+
+      // Reads kTest to buffer and checks if timeout and size are correct
+      if (count == 0)
+      {
+        CHECK(timeout == Esp8266::kReadTimeout);
+        CHECK(size == 68);
+
+        for (size_t i = 0; i < (sizeof(kTest)-1); i++)
+        {
+          data[i] = kTest[i];
+        }
+
+        count++;
+      }
+      // Checks FlushSerialBuffer has right timeout and size
+      else
+      {
+        CHECK(timeout == Esp8266::kReadTimeout);
+        CHECK(size == 1);
+      }
+
+      return Status::kTimedOut;
+    };
+    When(ConstOverloadedMethod(
+             mock_uart,
+             Read,
+             Status(uint8_t *, size_t, std::chrono::microseconds)))
+        .AlwaysDo(uart_response);
+    Esp8266::NetworkConnection_t network =
+      test_wifi_module.GetNetworkConnection();
+    // Checks to see if connection_info in NetworkConnection_t struct was
+    // written to
+    for (size_t i = 0; i < sizeof(kTest); i++)
+    {
+      CHECK(network.connection_info[i] == kTest[i]);
+    }
+    // Checks to see if GetNetworkConnectionInfoCommand was written to the
+    // esp8266
+    Verify(
+        ConstOverloadedMethod(mock_uart, Write, void(const uint8_t *, size_t))
+            .Using(Esp8266::kGetNetworkConnectionInfoCommand,
+              (sizeof(Esp8266::kGetNetworkConnectionInfoCommand)-1)));
   }
+
   SECTION("Connect to Access Point")
   {
+    static constexpr uint8_t kExpectedWrite[] =
+        "AT+CWJAP=\"ssid_test\",\"password_test\"\r\n";
```

Comment:
```suggestion
        R"(AT+CWJAP="ssid_test","password_test"\r\n)";
```
You can use a raw string here and get rid of the escape characters.

Reply:
For this one, I do need the escaped characters for \r and \n though.